lib/neopixel/blockingneopixel.h

```diff
@@ -0,0 +1,30 @@
+#include "mbed.h"
+
+/*
+    Data is written in G -> R -> B
+    Each byte is a number value from 0-255 of the intensity of that individual LED light
+*/
+
+class BlockingNeopixel
+{
+private:
+    int pixelNum;
+    int on_buffer[8] = {1,1,1,1,1,1,1,1};
+    int off_buffer[8] = {0};
+    
+public:
+    BlockingNeopixel(int numPixels);
+    ~BlockingNeopixel();
+    void pulse_1();
+    void pulse_0();
+    void writeByte(const int buffer[8]);
+    void showColour(char colour);
+    void blinkPixels(int flashes, char colour);
+    void writeAnyRGB(const int colour[3]);
+    void writeAnyRGBall(const int colour[3]);
+
+    //basic functions for autonomy
+    void flashGreen(int numFlashes, float delay_s);

```

Comment:
it'd be nice if there were explicit functions like solidBlue(), solidRed(). This would make the implementation code nicer and more intuitive i think. not too important though.